### ISSUE TITLE
500 Internal Server error on deployment

### ISSUE THREAD

**Issue by bhaveek424** [Elliott-Chong/quizmify#2]
Hi Elliot!
I am facing this error on the deployed app however ,API calls are working fine locally both in dev environment and after running `next build`.
![Screenshot from 2023-08-21 21-45-00](https://github.com/Elliott-Chong/quizmify/assets/61945842/9dc40072-028c-4d29-9509-501e8ee45846)
![Screenshot from 2023-08-21 21-47-13](https://github.com/Elliott-Chong/quizmify/assets/61945842/e8f29fae-946f-4f88-b839-b6919045c6a1)

What could be the reason for this? any idea?

**Comment by diyoifa** [comment]
it doesn't work for me locally
![image](https://github.com/Elliott-Chong/quizmify/assets/124792673/30191dc9-50a2-4a1b-8859-cf072d458625)

![image](https://github.com/Elliott-Chong/quizmify/assets/124792673/d8def2bc-f38c-4671-afa3-501ce422fe6e)

Any ideas that are wrong?

**Comment by bhaveek424** [comment]
@diyoifa you have to buy OpenAI API credits to make it work

**Comment by Elliott-Chong** [comment]
Yeah @diyoifa OpenAI might be rate limiting you for their API unless you buy some of their credits

**Comment by diyoifa** [comment]
oh thanks guys

**Comment by RichardWibowo** [comment]
> Hi Elliot! I am facing this error on the deployed app however ,API calls are working fine locally both in dev environment and after running `next build`. ![Screenshot from 2023-08-21 21-45-00](https://user-images.githubusercontent.com/61945842/262083208-9dc40072-028c-4d29-9509-501e8ee45846.png) ![Screenshot from 2023-08-21 21-47-13](https://user-images.githubusercontent.com/61945842/262083383-e8f29fae-946f-4f88-b839-b6919045c6a1.png)
> 
> What could be the reason for this? any idea?

did you find a fix for this ? i think its because the vercel rate limit but i dont think so

**Comment by anuranpaul** [comment]
> Hi Elliot! I am facing this error on the deployed app however ,API calls are working fine locally both in dev environment and after running `next build`. ![Screenshot from 2023-08-21 21-45-00](https://user-images.githubusercontent.com/61945842/262083208-9dc40072-028c-4d29-9509-501e8ee45846.png) ![Screenshot from 2023-08-21 21-47-13](https://user-images.githubusercontent.com/61945842/262083383-e8f29fae-946f-4f88-b839-b6919045c6a1.png)
> 
> What could be the reason for this? any idea?

Facing the same issue, what was the fix?

**Comment by RichardWibowo** [comment]
if 

> > Hi Elliot! I am facing this error on the deployed app however ,API calls are working fine locally both in dev environment and after running `next build`. ![Screenshot from 2023-08-21 21-45-00](https://user-images.githubusercontent.com/61945842/262083208-9dc40072-028c-4d29-9509-501e8ee45846.png) ![Screenshot from 2023-08-21 21-47-13](https://user-images.githubusercontent.com/61945842/262083383-e8f29fae-946f-4f88-b839-b6919045c6a1.png)
> > What could be the reason for this? any idea?
> 
> Facing the same issue, what was the fix?

if u already deployed it on vercel free tier, its because the gpt api is too long and you need to upgrade your vercel. other option is to host it elsewhere

**Comment by anuranpaul** [comment]
> > Facing the same issue, what was the fix?
> 
> if u already deployed it on vercel free tier, its because the gpt api is too long and you need to upgrade your vercel. other option is to host it elsewhere

Even with the pro trial tier, the questions.ts file is getting timed out due to the API. As per the video it's the same tier and that deployment worked fine?
